### ISSUE TITLE
RAF images from X-S10 appear massively cropped

### ISSUE THREAD

**Issue by dimk1** [Beep6581/RawTherapee#6013]
Hello,

some info on my setup:

**Camera**
The new X-S10 by Fujifilm. I think it is not included yet in Lensfun.

**RawTherapee**
```Version: 5.8-2339-g8af287a
Branch: dev
Commit: 8af287a
Commit date: 
Compiler: gcc 7
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V3.22.2
Lensfun: V0.3.2.0
Build type: release
Build flags:  -std=c++11  -Werror=unused-label -Werror=delete-incomplete -fno-math-errno -Wall -Wuninitialized -Wcast-qual -Wno-deprecated-declarations -Wno-unused-result -Wunused-macros -fopenmp -Werror=unknown-pragmas -O2 -DNDEBUG -ftree-vectorize
Link flags:   
OpenMP support: ON
MMAP support: ON
Build OS: Linux 4.4.0-101-generic x86_64
Build date: Sun, 12 Jul 2020 10:41:31 +0000 UTC
Build epoch: 1594550491
Build UUID: 4eb6c765-6983-487c-8d05-92b76a4c84c8
```

**System/OS**
```uname -a
Linux xps 5.8.0-7630-generic #32~1605108806~20.10~7e52b13-Ubuntu SMP Wed Nov 11 19:10:30 UTC  x86_64 x86_64 x86_64 GNU/Linux
```

**Reproduction**
1. Open the sample file [here](https://drive.google.com/file/d/1-ZvFh6qt4w3OCzBEIu2qWm-89NYBmrBV/view?usp=drivesdk) with the RT version shown above
2. Check that the image appears massively cropped (in fact only a small tile appears)

**Extra files** 
The respective pp3 file is [here](https://drive.google.com/file/d/1XXu5-jHjfakuRLB1ohrybqBm_vNVtKKn/view?usp=drivesdk).

**Comment by Thanatomanic** [comment]
@dimk1 Can you post a corresponding in-camera JPEG for reference?
This probably has nothing to do with lensfun, but more likely with the actual raw decoding.

**Comment by dimk1** [comment]
Sure, see [here](https://gofile.io/d/qu6829). Thanks

**Comment by dimk1** [comment]
Hi there, any updates on that?

**Comment by Thanatomanic** [comment]
Within 24 hours? Not likely. This is something @heckflosse needs to take a look at if he has the time.

**Comment by heckflosse** [comment]
@Thanatomanic Not much time atm. Maybe you can help out by porting this from ART
https://bitbucket.org/agriggio/art/commits/1a4cdaffe72bbc8b02351198102728c54809f625

**Comment by dimk1** [comment]
ART has the same problem as RT. I changed the compression algorithm in the camera and tried uncompressed RAW and compressed RAW (the example I submitted firstly was with Lossless RAW). The uncompressed RAW is processed as expected by RT, the compressed shows up like a completely black image.

**Comment by dimk1** [comment]
May I suggest to look at the same solution as with my previous bug on X-T30 [here](https://discuss.pixls.us/t/raf-files-from-xt-30-are-cropped-to-a-small-tile/17511)? It should be the same issue. The change should be in 
`
 rtengine/dcraw.cc`

shown in bold below

```
} else if (!strcmp(model, "X-Pro3") || !strcmp(model, "X-T3") || !strcmp(model, "X-T30") || !strcmp(model, "X-T4") || !strcmp(model, "X100V" || **!strcmp(model, "X-S10")** ) {
        width = raw_width = 6384;
        height = raw_height = 4182;
    }
```

**Comment by dimk1** [comment]
I did 2 things:

1. I downloaded the source code, did the change in the file I mentioned (rtengine/dcraw.cc) above and re-compiled.
2. I downloaded the source code for the lensfun, added the X-S10 in the file [here](https://github.com/lensfun/lensfun/blob/master/data/db/mil-fujifilm.xml) and re-compiled, namely:
```
    <camera>
        <maker>Fujifilm</maker>
        <model>X-S10</model>
        <mount>Fujifilm X</mount>
        <cropfactor>1.534</cropfactor>
    </camera>
```

Now the photos are not cropped and RT finds the new X-S10 entry from the lensfun. However, I noticed that in the top side of portrait pictures and right side of landscape pictures there is a slight black curved area. Any ideas to further troubleshoot this?
